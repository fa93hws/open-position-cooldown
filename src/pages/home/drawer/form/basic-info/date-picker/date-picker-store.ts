import { makeObservable, observable, action } from 'mobx';

export class DatePickerStore {
  date: Date;

  asDateString() {
    let month = `${this.date.getMonth() + 1}`;
    let day = `${this.date.getDate()}`;
    const year = this.date.getFullYear();

    if (month.length < 2) month = `0${month}`;
    if (day.length < 2) day = `0${day}`;

    return [year, month, day].join('-');
  }

  constructor(date = new Date()) {
    this.date = date;
    makeObservable(this, {
      date: observable.ref,
      setDate: action,
    });
  }

  setDate(val: Date | null) {
    if (val == null) {
      return;
    }
    this.date = val;
  }

  reset() {
    this.setDate(new Date());
  }
}
