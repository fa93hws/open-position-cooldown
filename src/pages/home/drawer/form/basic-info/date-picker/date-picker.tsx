import 'date-fns';
import * as React from 'react';
import { DatePicker, MuiPickersUtilsProvider } from '@material-ui/pickers';
import { observer } from 'mobx-react';

import { DatePickerStore } from './date-picker-store';

// module resolution in esbuild and jest is a bit different?
const DateFnsUtils = require('@date-io/date-fns'); // eslint-disable-line

export function createDatePicker(): [React.ComponentType, DatePickerStore] {
  const store = new DatePickerStore();
  const onChange = (val: Date | null) => store.setDate(val);
  const Component = observer(() => (
    <MuiPickersUtilsProvider utils={DateFnsUtils}>
      <DatePicker
        format="yyyy-MM-dd"
        value={store.date}
        onChange={onChange}
        autoOk
        disableFuture
        disableToolbar
        inputVariant="outlined"
        size="small"
      />
    </MuiPickersUtilsProvider>
  ));
  return [Component, store];
}
