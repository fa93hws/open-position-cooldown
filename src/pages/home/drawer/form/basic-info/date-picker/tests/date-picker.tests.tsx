import 'date-fns';
import * as React from 'react';

import { createDatePicker } from '../date-picker';

describe('DatePicker', () => {
  it('renders the following', () => {
    const [DatePicker, store] = createDatePicker();
    store.setDate(new Date('1999-02-04'));
    expect(<DatePicker />).toMatchRenderedSnapshot();
  });
});
