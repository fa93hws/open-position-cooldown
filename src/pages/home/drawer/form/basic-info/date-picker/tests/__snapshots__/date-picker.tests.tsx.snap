// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatePicker renders the following 1`] = `
<div
  class="MuiFormControl-root MuiTextField-root"
  format="yyyy-MM-dd"
>
  <div
    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
  >
    <input
      aria-invalid="false"
      class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
      readonly=""
      type="text"
      value="1999-02-04"
    />
    <fieldset
      aria-hidden="true"
      class="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
      style="padding-left:8px"
    >
      <legend
        class="PrivateNotchedOutline-legend-2"
        style="width:0.01px"
      >
        <span>
          â€‹
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;
